lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php

```diff
@@ -34,22 +33,14 @@ class AttributeDriver implements MappingDriver
 
     private readonly AttributeReader $reader;
 
-    /** @param array<string> $paths */
+    /**
+     * @param array<string> $paths
+     * @param bool          $reportFieldsWhereDeclared no-op, to be removed in 4.0
+     */
     public function __construct(array $paths, bool $reportFieldsWhereDeclared = false)
```

Comment:
```suggestion
     * @param true          $reportFieldsWhereDeclared no-op, to be removed in 4.0
     */
    public function __construct(array $paths, bool $reportFieldsWhereDeclared = true)
```

Also, let's throw if `false` has been passed explicitly.

Reply:
Or even better, let's use `func_num_args()`

Reply:
Let's switch to `func_get_args()` on 3.1.x